msp/mspimplvalidate.go

```diff
@@ -170,18 +170,15 @@ func (msp *bccspmsp) validateIdentityOUsV1(id *identity) error {
 }
 
 func (msp *bccspmsp) validateIdentityOUsV11(id *identity) error {
-	// Run the same checks as per V1
```

Comment:
There are 2 distinct OU validations:
1. [ouEnforcement](https://github.com/hyperledger/fabric/blob/a282f7132565d509d3ba6d502cd9c07cc5e567f0/msp/mspimplsetup.go#L253) means that the MSP classifies node types (peer, orderer, etc.) according to OUs. 
2. OUs that classify this MSP as defined [here](https://github.com/hyperledger/fabric/blob/a282f7132565d509d3ba6d502cd9c07cc5e567f0/msp/mspimplsetup.go#L390-L419).

The function [validateIdentityOUsV1](https://github.com/hyperledger/fabric/blob/f2dbb66542cf0520391c3388d7d2f1fb9662ff5f/msp/mspimplvalidate.go#L142) performs the second check. 

Before your proposed change, all identities had to comply to the second validation rule.

By doing the change you are proposing we are essentially saying that the first rule replaces the second one, but they can coexist. 

Aside for this, we cannot change the semantics of past version MSP validation functions such as this, because they have the potential to fork the network.
Consider a peer **peer0** that exists in the network for a year, and that has decided that a certain transaction is invalid because it didn't pass **validateIdentityOUsV1**, but it would have passed the other checks down this function. 
Now, imagine your commit is merged and the network is upgraded to the latest version of Fabric that includes your commit, and let **peer1** be a peer that joined to the network after this upgrade.
When this peer executes **validateIdentityOUsV11** for that transaction, if **ouEnforcement** is activated, the transaction would be deemed valid, and the state updates of that transaction would be applied on **peer1** in contrast to peer0, and we have reached a state fork.